dcplib/component_agents/analysis_agent.py

```diff
@@ -0,0 +1,100 @@
+import base64
+import logging
+import os
+import json
+
+import cromwell_tools
+from typing import List, Dict
+
+
+class AnalysisAgent:
+    def __init__(self) -> None:
+        """Interacts with the HCA DCP Pipelines Execution Service."""
+        deployment = os.environ['DEPLOYMENT_STAGE']
+        self.cromwell_url = 'https://cromwell.caas-prod.broadinstitute.org'
+        self.cromwell_collection = 'lira-int' if deployment == 'integration' else f'lira-{deployment}'
+        self.auth = cromwell_tools.cromwell_auth.CromwellAuth.harmonize_credentials(service_account_key=self.analysis_gcp_creds,
+                                                                                    url=self.cromwell_url)
+
+    @property
+    def analysis_gcp_creds(self):
+        # TODO: FIGURE OUT HOW TO PASS IN CREDS
```

Comment:
**NOTE**: Should NOT merge this PR until this is resolved. Will need some insights on this from Sam or Parth.

Reply:
Hi @rexwangcc so this is currently stored in aws secrets manager and retrieved via the config manager in dcplib

Reply:
Tbh I'm still not 100% sure how to do this since I've not used AWS secret manager 🤔 Shall I sub-class the `Config` and set up the analysis secrets on AWS secrets manager? Is there an existing example that I can follow?

Reply:
@parthshahva Could I get some help on this to close the PR?